__tests__/articles.test.js

```diff
@@ -51,6 +51,31 @@ describe("/api/articles", () => {
             });
           });
       });
+
+      it("should return articles filtered by topic when topic query is provided", () => {
+        return request(app)
+          .get("/api/articles?topic=coding")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body).toHaveProperty("articles");
+            expect(body.articles).toBeInstanceOf(Array);
+            body.articles.forEach((article) => {
+              expect(article.topic).toBe("coding");
+            });
+          });
+      });
+      it("should return all articles when a non-existent topic is provided", () => {
+        return request(app)
+          .get("/api/articles?topic=nonExistentTopic")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body).toHaveProperty("articles");
+            expect(body.articles).toBeInstanceOf(Array);
+            body.articles.forEach((article) => {
+              expect(article.topic).not.toBe("nonExistentTopic");
+            });
+          });
```

Comment:
This isn't quite the response we want.

If someone sends a valid query (which this is) but we don't have a topic with that name then we can do one of two things:
1. Your choice here, send back all the articles. But confuse the user as a result when they realise their query brought back articles with topics they didn't expect. They thought they were filtering by nonExistentTopic
2. The preferred choice, send back no articles. The request was valid but we just don't have any topics matching the search. It tells the user exactly what they need to know about their query